tests/tests_E0_evaluation.py

```diff
@@ -0,0 +1,76 @@
+import pickle
+import os
+import shutil
+import logging
+import mock
+
+import src.A0_initialization as initializing
+import src.B0_data_input_json as data_input
+import src.C0_data_processing as data_processing
+import src.D0_modelling_and_optimization as modelling
+import src.E0_evaluation as evaluation
+
+from .constants import TEST_REPO_PATH, INPUT_FOLDER
+
+PARSER = initializing.create_parser()
+TEST_INPUT_PATH = os.path.join(TEST_REPO_PATH, INPUT_FOLDER)
+TEST_OUTPUT_PATH = os.path.join(TEST_REPO_PATH, "MVS_outputs")
+
+DICT_BEFORE = os.path.join(TEST_REPO_PATH, "dict_values_before_E0.pickle")
+
+DICT_AFTER = os.path.join(TEST_REPO_PATH, "dict_values_after_E0.pickle")
+
+
+@mock.patch(
+    "argparse.ArgumentParser.parse_args",
+    return_value=PARSER.parse_args(["-i", TEST_INPUT_PATH, "-o", TEST_OUTPUT_PATH]),
+)
+def setup_module(m_args):
+    """Run the simulation up to module E0 and save dict_values before and after evaluation"""
+    if os.path.exists(TEST_OUTPUT_PATH):
+        shutil.rmtree(TEST_OUTPUT_PATH, ignore_errors=True)
+    user_input = initializing.process_user_arguments()
+
+    logging.debug("Accessing script: B0_data_input_json")
+    dict_values = data_input.load_json(
+        user_input["path_input_file"],
+        path_input_folder=user_input["path_input_folder"],
+        path_output_folder=user_input["path_output_folder"],
+        move_copy=False,
+    )
+    logging.debug("Accessing script: C0_data_processing")
+    data_processing.all(dict_values)
+
+    logging.debug("Accessing script: D0_modelling_and_optimization")
+    results_meta, results_main = modelling.run_oemof(dict_values)
+
+    with open(DICT_BEFORE, "wb") as handle:
+        pickle.dump(dict_values, handle, protocol=pickle.HIGHEST_PROTOCOL)
+
+    logging.debug("Accessing script: E0_evaluation")
+    evaluation.evaluate_dict(dict_values, results_main, results_meta)
+
+    with open(DICT_AFTER, "wb") as handle:
+        pickle.dump(dict_values, handle, protocol=pickle.HIGHEST_PROTOCOL)
+
+
+def test_evaluate_dict_append_new_fields():
+    with open(DICT_BEFORE, "rb") as handle:
+        dict_values_before = pickle.load(handle)
+
+    with open(DICT_AFTER, "rb") as handle:
+        dict_values_after = pickle.load(handle)
+
+    for k in ("kpi", "optimizedFlows"):
+        assert k not in dict_values_before
+        assert k in dict_values_after
```

Comment:
I took these two as they were the only one being only in the dict after.

> I could imagine that we try to gather a list of parameters which ALL have to be in the dict_values_after in the future.

I agree, I just didn't know which parameters had to be in `dict_values` :)

Reply:
You can also check "json_with_results.json" - there you can see the values currently added:

```
    "kpi": {
        "cost_matrix": "pandas dataframe",
        "scalar_matrix": "pandas dataframe",
        "scalars": {
            "annuity_om": 1499938.54592,
            "annuity_total": 1499976.46714,
            "costs_investment": 0.0,
            "costs_om": 20646400.82221,
            "costs_opex_fix": 2082877.0465600002,
            "costs_opex_var": 18563523.77566,
            "costs_total": 20646922.80138,
            "costs_upfront": 521.97917
        }
    },
    "optimizedFlows": {
        "ESS Li-Ion bus": "pandas dataframe",
        "Electricity (DSO) bus": "pandas dataframe",
        "Electricity bus": "pandas dataframe",
        "PV plant (mono) bus": "pandas dataframe"
    },
```

Maybe we can add multiple test functions for the different levels:
* one checking for kpi/optimizedFlows
* one checking the keys in kpi
* one checking the pd.dataframe in kpi/scalar_matrix
* one checking the pd.dataframe in kpi/cost_matrix
* one checking the keys in kpi/scalars

My question also referred to why you used 
```
 for k in ("kpi", "optimizedFlows"):
        assert k not in dict_values_before
        assert k in dict_values_after
```
instead of
```
assert "kpi" in dict_values_after
assert "optimizedFlows" in dict_values_after
```

Reply:
> My question also referred to why you used

Because I didn't know if the list would be larger in the future. I prefer add a key to the list than copy paste lines and change the key.

Reply:
Makes sense. If I have to change my tests checking for keys again, I will do the same.